### DISCUSSION TITLE
Databases which can be hosted on shared hosting

### DISCUSSION THREAD

**Discussion by wlatic** [vrtmrz/obsidian-livesync#149] [General]
A lot of people already have a website, which means they'll be paying per month or per year for shared hosting.

Is there a reason couchdb, in particular, was used? 

There are a few others (lower performance) like:
SleekDB
NeDB

Which would run on shared hosting, but I'm unsure if they'd give the correct level of performance. I'd be interesting in helping to adopt if they'd work.

**Comment by vrtmrz** [comment]
Thank you for asking me! I would like to explain why.

We are using CouchDB as a remote database and using PouchDB as a local database. And both of them can synchronise with each other easily.

As you said, CouchDB/PouchDB is not a popular database engine but also solves a big part of the difficulties of this plugin. Without this, Self-hosted LiveSync would not be made.

Notable advantages are like below:
- Is very good at synchronisation. (There is even a joke - "Bad at everything, except syncing") 
- Could be connected by the HTTP(S) efficiently. This helps synchronise on mobile devices.
- Is open.

So we rely on them for half part of this plugin.

---

Let me make an addition a bit more about the "open".

If we made a new server-side software, it would have to be audited. (Especially the people who are required to be self-hosted inside their network would need so)
Otherwise, everyone needs to trust our work in the server-side software. This might be terribly hard in total. 
Therefore, I have thought that the best way is to use solutions which are already audited by the company or the "Given enough eyeballs".

Of course, other implementations are also always welcome. 

We are able to make an adapter for the PouchDB to connect to the backend of your choice. (This might be one of the values of being open too)

If I received the pull request for the adapter which you made, I would love to configure that scheme as a connectable URI. (And this is why the remote-database configuration is not `URL`)

**Reply by wlatic** [reply]
Thank you for the indepth information. Makes total sense now.

I'm still looking around for something that'd really fulfil the role of couchDB to the same level, and at the same time run via shared hosting (generally node.js/php). Have now been using the software for 2 days after setting it up on Fly.io and everything is working great. Initially I was a little worried about "backup", but having the google drive / one drive watch the folder is enough for me for now!

Really great work on this and I'm hopeful at some point I can provide some help!
